cmd/tealdbg/local.go

```diff
@@ -531,23 +531,17 @@ func (r *LocalRunner) RunAll() error {
 		return fmt.Errorf("no program to debug")
 	}
 
-	configureDebugger := func(ep *logic.EvalParams) {
-		// Workaround for Go's nil/empty interfaces nil check after nil assignment, i.e.
-		// r.debugger = nil
-		// ep.Debugger = r.debugger
-		// if ep.Debugger != nil // FALSE
-		if r.debugger != nil {
-			ep.Tracer = logic.MakeEvalTracerDebuggerAdaptor(r.debugger)
-		}
-	}
-
 	txngroup := transactions.WrapSignedTxnsWithAD(r.txnGroup)
 	failed := 0
 	start := time.Now()
 
-	ep := logic.NewEvalParams(txngroup, &r.proto, &transactions.SpecialAddresses{})
-	ep.SigLedger = logic.NoHeaderLedger{}
-	configureDebugger(ep)
+	sep := logic.NewSigEvalParams(r.txnGroup, &r.proto, &logic.NoHeaderLedger{})
+	aep := logic.NewAppEvalParams(txngroup, &r.proto, &transactions.SpecialAddresses{})
+	if r.debugger != nil {
+		t := logic.MakeEvalTracerDebuggerAdaptor(r.debugger)
+		sep.Tracer = t
+		aep.Tracer = t
+	}
```

Comment:
better

Reply:
Yeah, the old concern about Go's weird nil handling of interfaces is gone, since we have to build the adaptor if non-nil.